poppy/wfe.py

```diff
@@ -353,6 +353,231 @@ def get_opd(self, wave):
         phase_screen = np.fft.ifftshift(np.fft.ifft2(np.fft.ifftshift(scaled))).real   # FT of scaled random PSD makes phase screen
 
         phase_screen -= np.mean(phase_screen)  # force zero-mean
-        opd = phase_screen / np.std(phase_screen) * self.wfe.to(u.m).value  # normalize to wanted input rms wfe
+        self.opd = phase_screen / np.std(phase_screen) * self.wfe.to(u.m).value  # normalize to wanted input rms wfe
+
+        return self.opd
+
+
+class PowerSpectrumWFE(WavefrontError):
+    r"""
+    WFE model specificed via a Power Spectral Density (PSD),
+    or a list of multiple PSDs, which follow von Karman PSD model:
+    
+    :math:`P(k) = \frac{\beta} {\left( \left(\frac{1}{L_{0}}\right)^{2} + |k|^{2} \right)^{{\alpha/2}}} e^{-(|k|l_{0})^{2}} + \beta_{sr}`
+    
+    where:
+    P: astropy quantity
+        Power Spectral Density at a spatial frequency.
+        Units: :math: `m^{2}m^{2}`
+        Assumes surface units of meters (first :math: `m^{2}`)
+    k: astropy quantity
+        Spatial frequency value, units 1/m
+    :math:`\alpha`: float 
+        The PSD index value 
+    :math:`\beta`: astropy quantity
+        The normalization constant. In units of :math: `\frac{m^{2}}{m^{\alpha-2}}`
+        Numerator assumes surface units of meters
+        Denominator assumes spatial frequency units are 1/m
+    :math:`L_{0}`: astropy quantity
+        The outer scale value, where the low spatial frequency flattens. Units: m
+    :math:`l_{0}`: float
+        Inner scale value, where the high spatial frequency flattens.
+    :math:`\beta_{sr}`: astropy quantity
+        Surface roughness normalization. Should match units of PSD.
+    
+    References:
+    Males, Jared. MagAO-X Preliminary-Design Review, 
+        Section 5.1: Optics Specifications, Eqn 1
+        https://magao-x.org/docs/handbook/appendices/pdr/
+    Lumbres, et al. In Prep.
 
-        return opd
+    Parameters
+    ----------
+    name : string
+        name of the optic
+    psd_parameters: list of lists
```

Comment:
`psd_parameters: list of lists`

Minor clarification.  Is this actually `psd_parameters: list (for single PSD), or list of lists (for multiple PSDs)`? I think so, from what you've written below, but want to check my understanding.

Reply:
Yes, this is correct.